test/e2e/clusterclass_rollout.go

```diff
@@ -186,7 +186,7 @@ func ClusterClassRolloutSpec(ctx context.Context, inputGetter func() ClusterClas
 			},
```

Comment:
Based on e2e-full we're checking for MD instead of MP somewhere in the new MP validations:
```
key "machinedeployment.clusters.x-k8s.io/revision" does not exist in map map[Cluster.topology.machinePool.annotation:Cluster.topology.machinePool.annotationValue ClusterClass.machinePool.annotation:ClusterClass.machinePool.annotationValue]
```

(I assume it's only surfaced on this PR, because now we're actually validating :))

Reply:
It looks like the error happens [here](https://github.com/willie-yao/cluster-api/blob/fix-mp-cc-e2e/test/e2e/clusterclass_rollout.go#L140), right after applying the cluster template. I think MachineDeployment should exist in this case so I'm wondering what happened here.

Reply:
Ah I think it's due to the lack of a revision annotation for MachinePools like it does for MachineDeployments. Removing this `without()` should fix it. https://github.com/willie-yao/cluster-api/blob/fix-mp-cc-e2e/test/e2e/clusterclass_rollout.go#L647

Reply:
Sounds good. Looks like we now hit another similar case
```
kubectl.kubernetes.io/last-applied-configuration
```

Reply:
@sbueringer Willie is out sick today so I pushed this change to the branch to speed up getting test signal in case we want to get this in before cutting the RC

Reply:
Thx!! I squashed & added co-authored-by